Slim/Formats/Playlists/CUE.pm

```diff
@@ -60,144 +122,590 @@ sub parse {
+			my($remCommand,$remValue)= _getRemCommandFromLine($value);
 
-			$cuesheet->{uc($1)} = $2;
+			if (!defined $remCommand or !defined $remValue){
 
-		} elsif ($line =~ /^(?:REM\s+)?(?:DISC|DISCNUMBER)\s+\"?([^\"]+)\"?/i) {
+				#No commads in rem, skipping;
 
-			# Some Cue sheets may not bother with quotes around a number
-			$cuesheet->{'DISC'} = $1;
+				my $msg = {};
+				$msg->{'message'}	= 'No commads in rem, skipping';
+				$msg->{'line'}		= $line;
+				$msg->{'command'}	= $command;
+				$msg->{'value'}		= $value;
+				$msg->{'remCommand'}	= $remCommand;
+				$msg->{'remValue'}	= $remValue;
 
-		} elsif ($line =~ /^(?:REM\s+)?(?:DISCC|DISCTOTAL|TOTALDISCS)\s+\"?([^\"]+)\"?/i) {
+				main::DEBUGLOG && $log->is_debug && $log->debug(Data::Dump::dump($msg));
 
-			# Some Cue sheets may not bother with quotes around a number
-			$cuesheet->{'DISCC'} = $1;
+			}elsif (!_isRemCommandAccepted($remCommand)){
 
-		} elsif ($line =~ /^(?:REM\s+)?(DATE)\s+(.*)/i) {
+				#Rem command refused;
 
-			# EAC CUE sheet has REM DATE not REM YEAR, and no quotes
-			$cuesheet->{'YEAR'} = $2;
+				my $msg = {};
+				$msg->{'message'}	= 'Rem command refused';
+				$msg->{'inAlbum'}	= $inAlbum;
+				$msg->{'currtrack'}	= $currtrack;
+				$msg->{'line'}		= $line;
+				$msg->{'command'}	= $command;
+				$msg->{'value'}		= $value;
+				$msg->{'remCommand'}	= $remCommand;
+				$msg->{'remValue'}	= $remValue;
 
-		} elsif ($line =~ /^(?:REM\s+)?(GENRE)\s+(.*)/i) {
+				main::DEBUGLOG && $log->is_debug && $log->debug(Data::Dump::dump($msg));
 
-			# Single worded GENRE doesn't have quotes
-			$cuesheet->{uc($1)} = $2;
+			} elsif ($remCommand eq 'END'){
 
-		} elsif ($line =~ /^(?:REM\s+)?(REPLAYGAIN_ALBUM_GAIN)\s+(.*)dB/i) {
+				if (!defined $currtrack){
+					#Ignored
+					main::DEBUGLOG && $log->debug("End found for missing Track");
 
-			$cuesheet->{uc($1)} = $2;
+				}elsif ($remValue =~ /^(\d+):(\d+):(\d+)/i) {
 
-		} elsif ($line =~ /^(?:REM\s+)?(REPLAYGAIN_ALBUM_PEAK)\s+(.*)/i) {
+					$tracks->{$currtrack}->{'END'} = ($1 * 60) + $2 + ($3 / 75);
 
-			$cuesheet->{uc($1)} = $2;
+				} elsif ($remValue =~ /^(.+)/i) {
 
-		} elsif ($line =~ /^FILE\s+\"(.*)\"/i) {
+					# Bug 11950, pass absolute end time in seconds (FLAC), since some loss of accuracy would
+					# occur if passing in MM:SS:FF format			
+					$tracks->{$currtrack}->{'END'} = $1;
+				}
 
-			$filename = $embedded || $1;
-			$filename = Slim::Utils::Misc::fixPath($filename, $baseDir);
-			
-			# Watch out for cue sheets with multiple FILE entries
-			$filesSeen++;
+			} elsif ($remCommand eq 'REPLAYGAIN_ALBUM_GAIN'){
 
-		} elsif ($line =~ /^FILE\s+\"?(\S+)\"?/i) {
+				if ($remValue =~ /^\"(.*)dB\"/i) {#"
 
-			# Some cue sheets may not have quotes. Allow that, but
-			# the filenames can't have any spaces in them.
-			$filename = $embedded || $1;
-			$filename = Slim::Utils::Misc::fixPath($filename, $baseDir);
-			
-			$filesSeen++;
+					($cuesheet, $tracks)=_addCommand($cuesheet, 
```

Comment:
All these calls to _addCommand()...

It's used to add valid standard commands (see line 414) and valid REM comments recognized as commands, normally without any adjustment (see line 331). 

For some commands and REM comments, some value check and adjustement is needed, of course I could reorganize the condition tree, store values and call _addCommand just once at the end, but I've tried to mantain previous structure whenever was possible, using a function in order to not replicate this bit of logic for each if/elsif branch. Arguable, I agree, but at runtime it's caled once per line, so it's just a matter of style.